### ISSUE TITLE
Native AOT publish ignores `DebugType=portable` 

### ISSUE THREAD

**Issue by vaind** [dotnet/sdk#35798]
### Describe the bug
Regardless of the `DebugType` setting, `dotnet publish` creates a windows-formatted (a.k.a. `DebugType=full`) PDB file.

AFAICT it is impossible to get a Portable PDB for a Native-AOT build. This format is necessary for offline stack trace symbolication with some services that only support Portable PDB, for example Sentry.

### To Reproduce

1. `dotnet new console --aot`
1. `dotnet publish`
2. Check the publish folder - it contains a full (Windows) PDB
3. add `<DebugType>portable</DebugType>` to the `.csproj`
1. `dotnet publish`
2. Check the publish folder - it contains a full (Windows) PDB

### Further technical details

<details>
<summary>` dotnet --info`</summary>
<pre>
.NET SDK:
 Version:   8.0.100-rc.1.23463.5
 Commit:    e7f4de8816

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.100-rc.1.23463.5\

.NET workloads installed:
 [ios]
   Installation Source: SDK 8.0.100-rc.1, VS 17.7.34031.279
   Manifest Version:    16.4.8825-net8-rc1/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.ios\16.4.8825-net8-rc1\WorkloadManifest.json
   Install Type:              Msi

 [maccatalyst]
   Installation Source: SDK 8.0.100-rc.1, VS 17.7.34031.279
   Manifest Version:    16.4.8825-net8-rc1/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.maccatalyst\16.4.8825-net8-rc1\WorkloadManifest.json
   Install Type:              Msi

 [maui-ios]
   Installation Source: VS 17.7.34031.279
   Manifest Version:    8.0.0-rc.1.9171/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.maui\8.0.0-rc.1.9171\WorkloadManifest.json
   Install Type:              Msi

 [maui-android]
   Installation Source: VS 17.7.34031.279
   Manifest Version:    8.0.0-rc.1.9171/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.maui\8.0.0-rc.1.9171\WorkloadManifest.json
   Install Type:              Msi

 [android]
   Installation Source: VS 17.7.34031.279
   Manifest Version:    34.0.0-rc.1.432/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.android\34.0.0-rc.1.432\WorkloadManifest.json
   Install Type:              Msi

 [maui-windows]
   Installation Source: VS 17.7.34031.279
   Manifest Version:    8.0.0-rc.1.9171/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.maui\8.0.0-rc.1.9171\WorkloadManifest.json
   Install Type:              Msi

 [maui-maccatalyst]
   Installation Source: VS 17.7.34031.279
   Manifest Version:    8.0.0-rc.1.9171/8.0.100-rc.1
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100-rc.1\microsoft.net.sdk.maui\8.0.0-rc.1.9171\WorkloadManifest.json
   Install Type:              Msi

Host:
  Version:      8.0.0-rc.1.23419.4
  Architecture: x64
  Commit:       92959931a3
  RID:          win-x64

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  6.0.317 [C:\Program Files\dotnet\sdk]
  6.0.414 [C:\Program Files\dotnet\sdk]
  7.0.401 [C:\Program Files\dotnet\sdk]
  8.0.100-rc.1.23463.5 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-rc.1.23421.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-rc.1.23419.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0-rc.1.23420.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</pre>

</details>

**Comment by reflectronic** [comment]
Yes, Native AOT does not support managed code debuggers. You can only debug Native AOT binaries with an unmanaged code debugger, which requires a full PDB containing unmanaged debugging info. (This is different from a full PDB that is produced by the C# compiler, which is still meant for managed code debuggers.)

**Comment by marcpopMSFT** [comment]
Old issue triage: Sounds like expected behavior. No plans to add a different error experience at this time.
